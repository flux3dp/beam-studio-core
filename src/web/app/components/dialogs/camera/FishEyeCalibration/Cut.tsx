import React, { useCallback } from 'react';
import { Button, Modal } from 'antd';

import deviceMaster from 'helpers/device-master';
import progressCaller from 'app/actions/progress-caller';
import useI18n from 'helpers/useI18n';

const PROGRESS_ID = 'fish-eye-calibration';

interface Props {
  onClose: (complete: boolean) => void;
  onBack: () => void;
  onNext: () => void;
}

const Cut = ({ onClose, onBack, onNext }: Props): JSX.Element => {
  const lang = useI18n();
  const handleNext = useCallback(async () => {
    progressCaller.openNonstopProgress({ id: PROGRESS_ID, message: 'Uploading Calibration Task' });
    try {
      await deviceMaster.doAdorCalibrationCut();
      onNext();
    } finally {
      progressCaller.popById(PROGRESS_ID);
    }
  }, [onNext]);
  return (
    <Modal
      open
      centered
      onCancel={() => onClose(false)}
      title={'tCut Calibration File'}
      footer={[
        <Button onClick={onBack} key="back">
          {lang.buttons.back}
        </Button>,
        <Button type="primary" onClick={handleNext} key="next">
          {lang.buttons.next}
        </Button>,
      ]}
      closable={false}
      maskClosable={false}
    >
      <div>
        Please put the material in the center of the workarea
      </div>
    </Modal>
  );
};

export default Cut;
