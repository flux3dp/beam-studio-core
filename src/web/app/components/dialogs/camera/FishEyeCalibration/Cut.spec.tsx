import React from 'react';
import { fireEvent, render } from '@testing-library/react';

import Cut from './Cut';

const mockDoAdorCalibrationCut = jest.fn();
jest.mock('helpers/device-master', () => ({
  doAdorCalibrationCut: () => mockDoAdorCalibrationCut(),
}));

const mockOpenNonstopProgress = jest.fn();
const mockPopById = jest.fn();
jest.mock('app/actions/progress-caller', () => ({
  openNonstopProgress: (...args) => mockOpenNonstopProgress(...args),
  popById: (id) => mockPopById(id),
}));

jest.mock('helpers/useI18n', () => () => ({
  buttons: {
    back: 'back',
    next: 'next',
  },
}));

const mockOnClose = jest.fn();
const mockOnBack = jest.fn();
const mockOnNext = jest.fn();

describe('test Cut', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const { baseElement } = render(<Cut onClose={mockOnClose} onBack={mockOnBack} onNext={mockOnNext} />);
    expect(baseElement).toMatchSnapshot();
  });

  test('onBack should work', () => {
    const { getByText } = render(<Cut onClose={mockOnClose} onBack={mockOnBack} onNext={mockOnNext} />);
    expect(mockOnBack).not.toBeCalled();
    fireEvent.click(getByText('back'));
    expect(mockOnBack).toBeCalledTimes(1);
  });

  test('onNext should work', async () => {
    const { getByText } = render(<Cut onClose={mockOnClose} onBack={mockOnBack} onNext={mockOnNext} />);
    expect(mockOpenNonstopProgress).not.toBeCalled();
    expect(mockDoAdorCalibrationCut).not.toBeCalled();
    mockDoAdorCalibrationCut.mockResolvedValueOnce(null);
    fireEvent.click(getByText('next'));
    expect(mockOpenNonstopProgress).toBeCalledTimes(1);
    expect(mockDoAdorCalibrationCut).toBeCalledTimes(1);

    expect(mockPopById).not.toBeCalled();
    expect(mockOnNext).not.toBeCalled();
    await new Promise((r) => setTimeout(r));
    expect(mockPopById).toBeCalledTimes(1);
    expect(mockOnNext).toBeCalledTimes(1);
  });
});
