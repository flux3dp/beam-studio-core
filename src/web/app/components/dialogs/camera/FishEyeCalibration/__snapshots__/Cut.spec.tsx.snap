// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Cut should render correctly 1`] = `
<body>
  <div />
  <div>
    <div
      class="ant-modal-root css-dev-only-do-not-override-hash"
    >
      <div
        class="ant-modal-mask"
      />
      <div
        class="ant-modal-wrap ant-modal-centered"
        tabindex="-1"
      >
        <div
          aria-labelledby="test-id"
          aria-modal="true"
          class="ant-modal css-dev-only-do-not-override-hash ant-zoom-appear ant-zoom-appear-prepare ant-zoom"
          role="dialog"
          style="width: 520px;"
        >
          <div
            aria-hidden="true"
            style="width: 0px; height: 0px; overflow: hidden; outline: none;"
            tabindex="0"
          />
          <div
            class="ant-modal-content"
          >
            <div
              class="ant-modal-header"
            >
              <div
                class="ant-modal-title"
                id="test-id"
              >
                tCut Calibration File
              </div>
            </div>
            <div
              class="ant-modal-body"
            >
              <div>
                Please put the material in the center of the workarea
              </div>
            </div>
            <div
              class="ant-modal-footer"
            >
              <button
                class="ant-btn css-dev-only-do-not-override-hash ant-btn-default"
                type="button"
              >
                <span>
                  back
                </span>
                
              </button>
              <button
                class="ant-btn css-dev-only-do-not-override-hash ant-btn-primary"
                type="button"
              >
                <span>
                  next
                </span>
                
              </button>
            </div>
          </div>
          <div
            aria-hidden="true"
            style="width: 0px; height: 0px; overflow: hidden; outline: none;"
            tabindex="0"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
